website/docs/opengl/c1/3-hello-texture.md

```diff
@@ -0,0 +1,388 @@
+the numeric values to color values to see what we are doing. In the shader, the values are being read like this:
+
+![UV values](../../../images/opengl/chapter1/leasson3/quad-with-uvs-and-numbers.png)
+
+The neat part is that the fragment shader already interpolates the coordinates! So we don't need to do that ourselves, and can just send a value per
+vertex. It saves a lot of work on our end!
+
+## Importing images as textures
+Now for the fun part: rendering an image!
+
+To start this section, first download
+[silk.png](https://github.com/dotnet/Silk.NET/blob/477154b99564aac9cf82426754857103fb88d2ea/examples/CSharp/OpenGL%20Tutorials/Tutorial%201.3%20-%20Textures/silk.png).
+We will use this image in this tutorial!
+
+In most cases with OpenGL, textures are uploaded as a sequence of bytes, this is usually done with a layout of 4 bytes per pixel (Red, Green, Blue, and Alpha).
+The hard part is: we can't just upload the bytes of a .png or .jpg file! These formats have a lot of unnecessary data, like headers and most importantly
+compression, which the GPU generally cannot understand.
+
+To load an image file as a byte array, we will first need a external library. In our case, we will use StbImageSharp.
+
+First of all, install the library using NuGet. To do so, run the following command:
+```
+dotnet add package StbImageSharp
+```
+
+After installing, add the following reference on the top of your code file:
+```c#
+using StbImageSharp;
+```
+
+At the top of your class, declare a `uint` to store the ID of the OpenGL texture object:
+```c#
+private static uint _texture;
+```
+
+Returning to your `OnLoad` method, add this line at the end of the method to create, and bind the texture.
+That `ActiveTexture` call is telling OpenGL that we are wanting to use the first texture slot.
+```c#
+_texture = _gl.GenTexture();
+_gl.ActiveTexture(TextureUnit.Texture0);
+_gl.BindTexture(TextureTarget.Texture2D, _texture);
+```
+
+After that, we need to load the image. You can do it with the following line:
+```c#
+// ImageResult.FromMemory reads the bytes of the .png file and returns all its information!
+ImageResult result = ImageResult.FromMemory(File.ReadAllBytes("silk.png"), ColorComponents.RedGreenBlueAlpha);
+```
+
+Now, with the image in memory, we need to upload the data to the GPU texture. To dot hat, we need a pointer for our bytes, the width, and the height of
+the texture. None of it is hardcoded, you get all this information inside `ImageResult`.
+```c#
+// Define a pointer to the image data
+fixed (byte* ptr = result.Data)
+    // Here we use "result.Width" and "result.Height" to tell OpenGL about how big our texture is.
+    _gl.TexImage2D(TextureTarget.Texture2D, 0, InternalFormat.Rgba, (uint)result.Width,
+        (uint)result.Height, 0, PixelFormat.Rgba, PixelType.UnsignedByte, ptr);
+```
+
+You don't need to understand what all these values mean! The OpenGL documentation explains it all in depth. Right now
```

Comment:
I disagree, you can't expect our users to blindly copy and paste code without understanding it at a surface level.

Reply:
Honestly I always just copy and paste this method to feed textures. This is very complex and full of weird parameters, also the size of the line makes this more complex to explain. I will try to provide some explanation anyway

Reply:
I agree with Perks, telling readers to just copy and paste code and not worry about understanding it can give a bad impression.

I've written an explanation to replace this paragraph in a suggestion below